Sliders.py

```diff
@@ -0,0 +1,302 @@
+import sys
+import numpy as np
+import xarray as xr
+from random import randint, sample
+from numpy import array
+from matplotlib.colors import LogNorm, Normalize
+
+from PyQt5.QtWidgets import QGridLayout, QRadioButton
+
+from matplotlib.backends.qt_compat import QtWidgets, is_pyqt5
+if is_pyqt5():
+    from matplotlib.backends.backend_qt5agg import (
+        FigureCanvas, NavigationToolbar2QT as NavigationToolbar)
+else:
+    from matplotlib.backends.backend_qt4agg import (
+        FigureCanvas, NavigationToolbar2QT as NavigationToolbar)
+from matplotlib.figure import Figure
+
+from Dimension import Dimension
+
+class ApplicationWindow(QtWidgets.QMainWindow):
+
+    def __init__(self):
+
+        super().__init__()
+
+        # Collection of letters used to create random dimension names
+        alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+        # Set number of dimension
+        self.n_dims = 7
+
+        # Create empty dictionary for slice selection
+        self.slice_selection = {}
+
+        # Generate random dimension names and sizes
+        self.dim_names = [alphabet[i] for i in sample(range(26),self.n_dims)]
+        self.dim_sizes = {self.dim_names[i]: randint(2,10) for i in range(self.n_dims)}
+
+        # Print dimension name and size
+        print("Dimension sizes: ")
+        print(" / ".join([self.dim_names[i] + ": " + str(self.dim_sizes[self.dim_names[i]]) for i in range(self.n_dims)]))
+
+        # Create a list of Dimension objects
+        self.dims = [Dimension(self.dim_names[i],self.dim_sizes[self.dim_names[i]],i) for i in range(self.n_dims)]
+
+        # Create a random n-D array
+        self.arr = np.random.rand(*[self.dim_sizes[key] for key in self.dim_names])
```

Comment:
Instead of creating data here, I suggest passing it as an argument to `__init__`, directly as a `DataArray` or `Dataset`.